src/main/java/com/github/pjfanning/xlsx/impl/StreamingWorkbookReader.java

```diff
@@ -202,29 +198,28 @@ private void loadSheets(Document workbookDoc, OoxmlReader reader, SharedStrings
     //Some workbooks have multiple references to the same sheet. Need to filter
     //them out before creating the XMLEventReader by keeping track of their URIs.
     //The sheets are listed in order, so we must keep track of insertion order.
-    OoxmlReader.OoxmlSheetIterator iter = reader.getSheetsData();
-    Map<PackagePart, InputStream> sheetStreams = new LinkedHashMap<>();
+    Iterator<OoxmlReader.SheetData> iter = reader.sheetIterator();
+    List<PackagePart> sheetParts = new ArrayList<>();
```

Comment:
Looking at the comments above, I think the original implementation was using a `LinkedHashMap` to filter out duplicate references to the same sheet (while maintaining insertion order).

If that's the case, then maybe a `LinkedHashSet` should be used instead of an `ArrayList` (or have you handled this is `OoxmlReader`)?